src/Fixer/Whitespace/NoExtraBlankLinesFixer.php

```diff
@@ -370,7 +370,20 @@ private function removeBetweenUse($index)
         }
 
         $nextUseCandidate = $this->tokens->getNextMeaningfulToken($next);
-        if (null === $nextUseCandidate || 1 === $nextUseCandidate - $next || !$this->tokens[$nextUseCandidate]->isGivenKind($this->tokens[$index]->getId())) {
+        if (null === $nextUseCandidate || !$this->tokens[$nextUseCandidate]->isGivenKind($this->tokens[$index]->getId())) {
+            return;
+        }
+
+        $isOnSameLine = true;
+        for ($i = $nextUseCandidate; $i > $index; --$i) {
+            if (false !== strpos($this->tokens[$i]->getContent(), "\n")) {
```

Comment:
Do we need to handle the case where there are only slash r characters for newlines?

Reply:
we could, so we should go for a regex solution using `\R`?

Reply:
I think so, yeh. There are probably some other places within the codebase that suffer this bug too. It might be worth adding a special method into the token class for testing for newline, to avoid this class of bug.

Reply:
I've updated to use regex, however I didn't find other case for now (did a quick check only though), so I get it within the class.